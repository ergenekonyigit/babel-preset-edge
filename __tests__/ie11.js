import buildPreset from "../src"
import { fixtures, titles, check } from "./core"

describe("IE11", () => {
  const options = buildPreset(null, {
    modules: false,
    target: { browsers: "ie 11" },
    sourceMaps: false
  })

  fixtures.forEach((fileName, index) => {
    test(titles[index], () => check(fileName, options))
  })
})
