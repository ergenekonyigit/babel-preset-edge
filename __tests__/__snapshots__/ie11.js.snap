// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`es2015: classes 1`] = `
"import _classCallCheck from \\"babel-runtime/helpers/classCallCheck\\";

var Component = function () {
  function Component() {
    _classCallCheck(this, Component);
  }

  Component.prototype.main = function main() {};

  return Component;
}();"
`;

exports[`es2015: generator 1`] = `
"function* idMaker() {
  var index = 0;
  while (true) {
    yield index++;
  }
}"
`;

exports[`es2015: set 1`] = `"new Set([1, 2, 3]);"`;

exports[`es2016: exponentiation 1`] = `"var result = Math.pow(2, 3);"`;

exports[`es2017: async await 1`] = `
"import _default from \\"nodent-runtime\\";
function hello() {
  return new Promise(function ($return, $error) {
    return $return();
  }.$asyncbind(this));
}"
`;

exports[`esnext: class properties 1`] = `
"import _classCallCheck from \\"babel-runtime/helpers/classCallCheck\\";

var Component = function Component() {
  _classCallCheck(this, Component);

  this.onClick = function () {};
};"
`;

exports[`esnext: import 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

console.log(_camelCase(\\"hello world\\"));"
`;

exports[`esnext: object spread 1`] = `
"var original = { bar: 10 };
var variant = Object.assign({ foo: 1 }, original);"
`;

exports[`lodash: async 1`] = `
"import _parallel from \\"async/parallel\\";

_parallel([]);"
`;

exports[`lodash: lodash 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

_camelCase(\\"hello world\\");"
`;

exports[`lodash: recompose 1`] = `
"import _compose from \\"recompose/compose\\";

_compose([]);"
`;

exports[`react: jsx 1`] = `
"var _jsxFileName = \\"./__tests__/__fixtures__/react_jsx.js\\";
function render() {
  return React.createElement(
    \\"h1\\",
    {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 2
      },
      __self: this
    },
    \\"Hello\\"
  );
}"
`;

exports[`react: jsx spread 1`] = `
"var _jsxFileName = \\"./__tests__/__fixtures__/react_jsx-spread.js\\";
function render(props) {
  return React.createElement(
    \\"h1\\",
    Object.assign({ className: \\"large\\" }, props, {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 3
      },
      __self: this
    }),
    \\"Hello\\"
  );
}"
`;
